### PR TITLE
add repository to search results page breadcrumbs

### LINKED ISSUE #708: Update list view of search results - B - Result item breadcrumb
Split out of #669 

Screen Shot 2019-08-28 at 4 36 48 PM

**B: Result item breadcrumb** - This is similar to the breadcrumb shown in the current search results display. What's changed: 

- [x] Add the repository to the beginning of the breadcrumb
- [x] Move the breadcrumb so it is the last row of the result item entry
- [x] Change the link styling to remove the link color. The breadcrumb elements should still be individually linked as they are now, we're just trying to reduce the amount of blue (future styling updates TBD).

### PR DESCRIPTION
**Pull request by labradford**
closes #708 Ref #669 
### Issue
Add the repository to the beginning of the breadcrumb
Move the breadcrumb so it is the last row of the result item entry
Change the link styling to remove the link color. 

### Solution
- Add repository (and link to it if available) to the parents_to_links method in arclight_helper
- Move breadcrumbs to last item on the search index (this along with indenting items under the title required a lot of UI changes)
- Changed the link style in search_results.scss

### Discussion
Because of the new design, I ended up creating a new index file for the search results. The layout for bookmarks, online content, and container are not styled purposely. Those are addressed on separate tickets.

### Demo
<a href="https://cl.ly/d0534aef08c4" target="_blank"></a>

### CODE REVIEW

File: app/helpers/arclight_helper.rb [hunk 1]
```diff
@@ -225,4 +231,15 @@ def generic_should_render_field?(config_field, document, field)
   def generic_render_document_field_label(config_field, document, field: field_name)
     send(:"render_document_#{config_field}_label", document, field: field)
   end
+
+  private
+
+  def build_repository_link(document)
+    repository_path = Arclight::Repository.find_by(name: document.repository)&.slug if document.repository.present?
```
[the remark this answers is not in the document]
Comment (labradford) [reply app/helpers/arclight_helper.rb]: No, I don't think so. document.repository_config returns nil.

Comment (mejackreed) [inline app/helpers/arclight_helper.rb]:
```suggestion
    repository_path = document.repository_config
```
I'm wondering if this is equivalent?

Comment (labradford) [reply app/helpers/arclight_helper.rb]: ah, yes. I see what I did now. this does work. updated and pushed change

[the remark this answers is not in the document]
Comment (labradford) [reply app/helpers/arclight_helper.rb]: never mind. repository_config does work

File: app/assets/stylesheets/arclight/modules/search_results.scss [hunk 2]
```diff
@@ -109,3 +101,7 @@
 .col-no-right-padding {
   padding-right: 0 !important;
 }
+
+.breadcrumb-links a {
+  color: #212529;
```
Comment (mejackreed) [inline app/assets/stylesheets/arclight/modules/search_results.scss]: Can we use the Bootstrap variable here `$body-color`?

File: app/helpers/arclight_helper.rb [hunk 3]
```diff
@@ -6,9 +6,15 @@ module ArclightHelper
   ##
   # @param [SolrDocument]
   def parents_to_links(document)
-    safe_join(document_parents(document).map do |parent|
+    breadcrumb_links = []
+
+    breadcrumb_links << build_repository_link(document)
+
+    breadcrumb_links << Arclight::Parents.from_solr_document(document).as_parents.map do |parent|
```
Comment (mejackreed) [inline app/helpers/arclight_helper.rb]:
```suggestion
    breadcrumb_links << document_parents(document).map do |parent|
```
Can we reuse `document_parents(document)` here?

File: app/helpers/arclight_helper.rb [hunk 4]
```diff
@@ -225,4 +231,15 @@ def generic_should_render_field?(config_field, document, field)
   def generic_render_document_field_label(config_field, document, field: field_name)
     send(:"render_document_#{config_field}_label", document, field: field)
   end
+
+  private
+
+  def build_repository_link(document)
+    repository_path = document.repository_config&.slug
+    if repository_path.present?
```
Comment (mejackreed) [inline app/helpers/arclight_helper.rb]: Do we need to check this? My understanding was that `slug` is a required property of `Arclight::Repository`

Comment (labradford) [reply app/helpers/arclight_helper.rb]: in the fixtures, the large collection sample doesn't have a slug so this did not work without the check.

File: app/assets/stylesheets/arclight/modules/search_results.scss (hunk 2 above)
Comment (labradford) [reply app/assets/stylesheets/arclight/modules/search_results.scss]: Updated to use variable

File: app/helpers/arclight_helper.rb (hunk 3 above)
Comment (labradford) [reply app/helpers/arclight_helper.rb]: updated per your request to use document_parents(document)

File: app/helpers/arclight_helper.rb (hunk 4 above)
Comment (mejackreed) [reply app/helpers/arclight_helper.rb]: Oh weird, ok!

**Pull request merged**